internal/api/v1beta1/permission_check.go

```diff
@@ -15,13 +15,96 @@ import (
 	"google.golang.org/grpc/status"
 )
 
+type resourcePermissionResult struct {
+	objectId        string
+	objectNamespace string
+	permission      string
+	allowed         bool
+}
+
 func (h Handler) CheckResourcePermission(ctx context.Context, req *shieldv1beta1.CheckResourcePermissionRequest) (*shieldv1beta1.CheckResourcePermissionResponse, error) {
 	logger := grpczap.Extract(ctx)
 	//if err := req.ValidateAll(); err != nil {
 	//	formattedErr := getValidationErrorMessage(err)
 	//	logger.Error(formattedErr.Error())
 	//	return nil, status.Errorf(codes.NotFound, formattedErr.Error())
 	//}
+	//  To have backward compatibility
+	if req.ObjectId != "" && len(req.ResourcePermissions) == 0 {
+		return h.checkSingleResourcePermission(ctx, req)
```

Comment:
if len==0, then why `checkSingleResourcePermission`?

Reply:
Only when both objectID and the length of the `ResourcePermissions` list are zero, we are allowing for `checkSingleResourcePermission` to ensure backward compatibility.

If the len is non-zero, then the client would have populated the list and we will go with the new contract.

Reply:
As discussed, we will be proceeding with the backward compatibility for now.